packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts

```diff
@@ -84,7 +85,12 @@ export const useRunMetricExplorerQuery = ({
             !!exploreName &&
             !!metricName &&
             !!comparison &&
-            !!dateRange,
+            !!dateRange &&
+            (comparison.type !== MetricExplorerComparison.DIFFERENT_METRIC ||
+                (comparison.type ===
+                    MetricExplorerComparison.DIFFERENT_METRIC &&
```

Comment:
only enable the api request for different metric if all required params are set

Reply:
my suggestion here is to have query params passed to this hook and then use `enabled` prop to control it from the UI component

Reply:
thoughts on this? https://github.com/lightdash/lightdash/pull/12765/commits/b4ec874420b368ee8ccac435bcbf446562e5aefc